inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -98,12 +98,83 @@ def article_exists(obj, eng):
     """
     matches = dedupe_list(match(obj.data))
     record_ids = [el['_source']['control_number'] for el in matches]
```

Comment:
Since https://github.com/inspirehep/inspire-matcher/commit/3d9cec0df0944acd7733949e776f1e95248a13be one can avoid this and just pass the `source` parameter to the `match` call to do filtering already on ES (which is much faster, as it reduces network communication).

Reply:
The filtering works as expected, but it still returns a minimal object, from which we have to extract only the `control_number`, so this statement is still required.

This is the object returned by `match`:
```python
{
  u'_id': u'c672d49e-f633-45dd-89d3-78d653098350',
  u'_index': u'records-hep',
  u'_score': 1.292094,
  u'_source': {u'control_number': 1234},
  u'_type': u'hep'
}
```